### ISSUE TITLE
Suggested clarification in config.ini.php

### ISSUE THREAD

**Issue by freibadschwimmer** [AlfnRU/roundcube-password_recovery#9]
Hi,
first of all, thanks a lot for this very useful plugin, great work!

When setting up, I was trying to follow the instructions from the config.ini.php, which ask for
`// Database connection string and table name with user passwords`
In my case, this would be the postfix database; However, after running into constant error messages, I changed the settings to connect the password_recovery plugin with the roundcube database (which does not have any passwords stored), and it finally worked.
So, it seems that the guidelines here are confusing, and one has to actually connect the plugin with the roundcube database. I think it would be good to clarify this in the config.ini.php.

My setup:
password_recovery 1.2
password 5.3
Roundcube 1.6.0

**Comment by AlfnRU** [comment]
Hi!
The connection string must be to the database where the user data is stored. The name of the table with user data is specified in the "pr_users_table" parameter. The fields that are used by the plugin are listed in the "pr_fields" parameter. The plugin checks and adds (if necessary) the fields listed in the "pr_fields" parameter to this table the first time it starts. And two more fields are 'token' and 'token_validity'.
The plugin also uses the fields 'user _ id', 'username', 'email' and 'password'.

All this works for me with the postfix database. Structure of the mailbox table:

> CREATE TABLE `mailbox` (
  `username` varchar(255) NOT NULL,
  `password` varchar(255) NOT NULL,
  `name` varchar(255) NOT NULL,
  `maildir` varchar(255) NOT NULL,
  `quota` bigint(20) NOT NULL DEFAULT 0,
  `local_part` varchar(255) NOT NULL,
  `domain` varchar(255) NOT NULL,
  `created` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `modified` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `active` tinyint(1) NOT NULL DEFAULT 1,
  `phone` varchar(30) NOT NULL DEFAULT '',
  `email_other` varchar(255) NOT NULL DEFAULT '',
  `password_expiry` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `token` varchar(255) DEFAULT '',
  `token_validity` datetime DEFAULT '2000-01-01 00:00:00',
  `question` varchar(255) DEFAULT '',
  `answer` varchar(255) DEFAULT ''
)

**Comment by freibadschwimmer** [comment]
I guess, the issue in my case is that the database structure differs (and it might for many users).

I noticed, that while the pr plugin wrote the new columns (token, phone, question etc) into the database table with the password, it could not load any data due to the missing column "username" - which in my case is in the roundcube database (which uses the email address to authenticate against the postfix database; hence the postfix db has no column username).
The way I configured the pr plugin, it uses the password plugin for the password changes (which I thought was obligatory?).
The password plugin has its own configuration pointing to the database and table that stores the passwords (with "password_db_dsn").

So maybe it would make sense to clarify in the pr/config.ini.php that if the password plugin is used for the password management, the pr plugin needs to point to the table that holds the (roundcube) usernames? Maybe something like

```
// Database connection string and table name with user passwords
// If you have set pr_use_password_plugin = true, then connect to the database and table with the username,
// and make sure that the password plugin is properly configured to connect to the db that stores the passwords
```
